### ISSUE TITLE
hostfile deprecated without warning

### ISSUE THREAD

**Issue by willthames** [ansible/ansible#25160]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
configuration

##### ANSIBLE VERSION
```
ansible 2.4.0 (devel 06f76a2741) last updated 2017/05/30 21:47:18 (GMT +1000)
  config file = 
  configured module search path = [u'/Users/will/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/will/src/ansible/lib/ansible
  executable location = /Users/will/src/ansible/bin/ansible
  python version = 2.7.13 (default, Dec 18 2016, 07:03:39) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
```

##### CONFIGURATION
[defaults]
hostfile = ./inventory

##### OS / ENVIRONMENT
N/A

##### SUMMARY
While I appreciate that hostfile is a rubbish name for the inventory configuration, there have been zero warnings that hostfile is about to be deprecated, and now it just fails silently on configuration that worked without warnings in 2.3

I think there should at least be a message saying "It looks like you're using hostfile in ansible.cfg. Please use inventory instead"

##### STEPS TO REPRODUCE
Create an ansible.cfg with hostfile = ./inventory

##### EXPECTED RESULTS
Some kind of warning/error that hostfile is no longer a valid configuration item

##### ACTUAL RESULTS
```
 [WARNING]: No inventory was parsed, only implicit localhost is available

 [WARNING]: provided hosts list is empty, only localhost is available
```

**Comment by bcoca** [comment]
we need to add a facility for configuration to notify on deprecated and removed settings. Since we were not able to do this when we deprecated host file .. it has gone under the radar.

There are notes in changelog .. but that does not mean a runtime notification would not be much nicer.

**Comment by bcoca** [comment]
fixed in several commits, reinstated the option and used the facility described above to emit a deprecation message IF set.

**Comment by zorun** [comment]
For reference, the fix has been pushed to the stable-2.4 branch so the next 2.4.x release should be fine (see 63b4e41d0cb37634b9edce9de469817832413d49)

**Comment by torfsen** [comment]
Since the deprecation message doesn't tell you how to solve the issue: for anyone running into

> [DEPRECATION WARNING]: [defaults]hostfile option, The key is misleading as it can also be a list of hosts, a directory or a list of paths . This feature will be removed in version 2.8.

The solution is to replace the `hostfile` option in the `defaults` section of your `ansible.cfg` with the new [`inventory` option](http://docs.ansible.com/ansible/latest/intro_configuration.html#inventory-file).

**Comment by willthames** [comment]
We could add that to the deprecation message.

**Comment by willthames** [comment]
@torfsen thanks very much for that comment - there was already a mechanism in place for this, but it had a bug that meant that it wasn't working for any deprecation message. 

I've made a fix in #34672
  
With that fix, you get:

```
$ ansible-playbook playbook.yml 
[DEPRECATION WARNING]: [defaults]hostfile option, The key is misleading as it can also be a list of hosts, a directory or a list of paths , use [defaults] inventory=/path/to/file|dir 
instead. This feature will be removed in version 2.8. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
```

**Comment by malutanpetronel** [comment]
and what I suppose to put in the hosts file ?
use [defaults] inventory=/path/to/file|dir ?

I have:

# default
# localhost ansible_connection=local

[all:children]
group

[group]
123.1.21.1
